woocommerce/woocommerce-gutenberg-products-block
# Refactor logic for handling active payment method with express payment methods via checkout

Fixes: #2168 

This pull addresses a couple things:

## Handling setting status and active payment method via checkout for express payment methods.

While this improves things so express payment methods don't have to worry about state changes etc in checkout, they still will need to implement the provided `onClick` and `onClose` handlers provided via checkout. 

## Explicitly provide a `paymentMethodId` that is separate from `activePaymentMethod`

The configuration object for registered payment methods now includes a `paymentMethodId` property and the existing `id` property was renamed to `name`.  This new property allows for payment methods to indicate a value to use in the request as the payment method identifier when it's different than the `name` property (which is what it will default to if `paymentMethodId` is not provided).  

I also improved some of the stripe server side things so that we only modify ordermeta for stripe payment_request packages.

## To test

This impacts all payment method processing so testing needs to cover cheque, stripe cc, and stripe payment request (express payment) purchases.  Ensure the flow works as expected and there are no console errors. Also ensure the order details show the correct expected values.
